### PR TITLE
Add a list of methods and field types allowed in FormField

### PR DESCRIPTION
**Pull request by Klemart3D**

### CODE REVIEW

**Comment by yannicka** [comment]
In the long term, won't the `FormField` class disappear in favour of Symfony forms?

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 1]
```diff
@@ -5,3 +5,102 @@ chapter: true
 ---
 
 # Forms
+
+From PrestaShop 1.7, fields are managed in FormField class and display is set in themes/<themeName>/templates/_partials/form-fields.tpl template.
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:9]:
```suggestion
From PrestaShop 1.7, fields are managed in FormField class and display is set in `themes/<themeName>/templates/_partials/form-fields.tpl` template.
```

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 2]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+* **setLabel**: Set the label of the field
+* **setValue**: Set the field value
+* **setAvailableValues**: Set the available values for the field, to select among other things
+* **addAvailableValue**: Add an available value for the field
+* **setMaxLength**: Maximum length of the field
+* **setConstraint**: Add a validation constraint to the field, call methods are those of the Validation class (ex isEmail)
+
+Here are all the possible types of fields:
+
+```
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:25]:
```suggestion
```php
```

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 3]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+return [
+    // Standard text field
+    (new FormField)
+        ->setName('first_name')
+        ->setType('text')
+        ->setRequired(true)
+        ->setValue("John")
+        ->setMaxLength("128")
+        ->setLabel($this->l('Your first name')),
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:34]:
```suggestion
        ->setLabel($this->trans('Your first name')),
```
We should avoid the `$this->l` calls :thinking:

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 4]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+    (new FormField)
+        ->setName('file_upload')
+        ->setType('file')
+        ->setLabel($this->l('Upload a document')),
+        
+    // Select field
+    (new FormField)
+        ->setName('select_field')
+        ->setType('select')
+        ->setAvailableValues(array('key' => 'value 1', 'key2' => 'value2'))
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:46]:
```suggestion
        ->setAvailableValues(['key' => 'value 1', 'key2' => 'value2'])
```

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 5]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+        ->setName('checkbox_field[]')
+        ->setType('checkbox')
+        ->setValue(1)
+        ->setLabel($this->l('Checkbox type')),
+        
+    // Radio buttons field
+    (new FormField)
+        ->setName('radio_field')
+        ->setType('radio-buttons')
+        ->setAvailableValues(array('key' => 'value 1', 'key2' => 'value2'))
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:67]:
```suggestion
        ->setAvailableValues(['key' => 'value 1', 'key2' => 'value2'])
```

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 6]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+    // Country select field (like select but add a 'js js-country' class)
+    (new FormField)
+        ->setName('country_field')
+        ->setType('countrySelect')
+        ->setAvailableValues(['key' => 'value 1', 'key2' => 'value2'])
+        ->setLabel($this->l('Country select')),
+        
+    // Checkbox field
+    (new FormField)
+        ->setName('checkbox_field[]')
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:58]:
```suggestion
        ->setName('checkbox_field')
```

Is it really needed to add `[]`?

File: src/content/1.7/modules/concepts/forms/_index.md [hunk 7]
```diff
@@ -5,3 +5,102 @@ chapter: true
...
+    // Birthday field
+    (new FormField)
+        ->setName('birthday_field')
+        ->setType('birthday')
+        ->setLabel($this->l('Birthday')),
+        
+    // Password field
+    (new FormField)
+        ->setName('password_field')
+        ->setType('password_')
```
Comment (PierreRambaud) [inline src/content/1.7/modules/concepts/forms/_index.md:85]:
```suggestion
        ->setType('password')
```
underscore?

**Comment by Klemart3D** [comment]
Thanks for the review, I committed suggested modifications.

**Comment by Progi1984** [comment]
Thanks @Klemart3D for your modifications.

**Comment by matks** [comment]
Thank you @Klemart3D

**Comment by nenes25** [comment]
@Klemart3D it's nice to share and translate documentation with everyone.
But when you copy/paste an article a good thing could  be to notify me at least about the fixed errors, so i can fix them on my side too

**Pull request merged**